internal/gitinterface/sync.go

```diff
@@ -72,10 +72,25 @@ func FetchRefSpec(ctx context.Context, repo *git.Repository, remoteName string,
 	}
 
 	err = remote.FetchContext(ctx, fetchOpts)
-	if errors.Is(err, transport.ErrEmptyRemoteRepository) || errors.Is(err, git.NoErrAlreadyUpToDate) {
-		return nil
+	if err != nil {
+		if errors.Is(err, transport.ErrEmptyRemoteRepository) || errors.Is(err, git.NoErrAlreadyUpToDate) {
+			return nil
+		}
+		return err
 	}
-	return err
+
+	ref, err := repo.Reference(plumbing.HEAD, false)
+	if err != nil {
+		return err
+	}
+
+	wt, err := repo.Worktree()
+	if err != nil {
+		return err
+	}
+	return wt.Checkout(&git.CheckoutOptions{
```

Comment:
I don't think we want to do this - fetch typically doesn't modify the local workspace, and you don't want a checkout to the last HEAD to possibly undo uncommitted changes.

Reply:
I wasn't sure what to do here. Note that this is supposed to be the behaviour of pull, so fetch + merge. Without the checkout, the worktree was left unchanged so git assumed there were staged changes (a revert of whatever was just fetched).

Reply:
I've pushed an attempt at separating Pull from Fetch in gitinterface. The behavior is still incorrect, too aggressive with the worktree.Reset even when the mode is set to MergeReset.

Reply:
@wlynch Would it make sense to dial the scope back here to focusing only on the gittuf namespaces? We have clearer rules on how those branches should work (for example, rsl is always fast forward), and we don't have to mess with worktrees at all. For now, the user'd have to do git push like normal and gittuf sync, and we work out how to clean this up using #4.